### PR TITLE
Form: label template

### PR DESCRIPTION
**Pull request by EugeniyKiyashko**

### CODE REVIEW

File: js/ui/form/ui.form.layout_manager.js [hunk 1]
```diff
@@ -560,6 +563,13 @@ const LayoutManager = Widget.inherit({
         }
 
         const name = item.dataField || item.name;
+        const formOrLayoutManager = this._getFormOrThis();
+
+        const onLabelsRendered = () => {
+            if(formOrLayoutManager.option('templatesRenderAsynchronously')) {
+                isLastFieldItem && formOrLayoutManager._alignLabels(this, this.isSingleColumnMode(formOrLayoutManager));
```
Comment (ksercs) [inline js/ui/form/ui.form.layout_manager.js]:
```suggestion
                formOrLayoutManager._alignLabels(this, this.isSingleColumnMode(formOrLayoutManager));
```

File: js/ui/form/components/label.js [hunk 2]
```diff
@@ -37,6 +37,9 @@ export function renderLabel({ text, id, location, alignment, labelID = null, mar
         labelTemplate.render({
             container: getPublicElement($labelContent),
             model: labelTemplateData,
+            onRendered() {
+                onLabelsRendered?.();
```
Comment (ksercs) [inline js/ui/form/components/label.js]: Are u sure that this `onRendered` will be executed the latest? What if we have 2 items in the dataSource, and labelTemplate for the first one renders later then the second one.

Comment (EugeniyKiyashko) [reply js/ui/form/components/label.js]: https://github.com/DevExpress/DevExtreme/pull/22961/commits/1f68deb5b4165de0274328699c1969b626b6a1af

**Comment by ksercs** [comment]
@EugeniyKiyashko  can we make that PR's name more meaningful?

File: js/ui/form/components/field_item.js [hunk 3]
```diff
@@ -77,7 +77,9 @@ export function renderFieldItem({
     let $label = null;
 
     if(needRenderLabel) {
-        labelOptions.labelTemplateData = getTemplateData(item, editorOptions, formOrLayoutManager);
+        if(labelOptions.labelTemplate) {
+            labelOptions.labelTemplateData = getTemplateData(item, editorOptions, formOrLayoutManager);
```
Comment (ksercs) [inline js/ui/form/components/field_item.js:81]: We can write it without condition
`labelOptions.labelTemplateData ??= getTemplateData(item, editorOptions, formOrLayoutManager);`

File: js/ui/form/ui.form.layout_manager.js [hunk 4]
```diff
@@ -320,16 +320,26 @@ const LayoutManager = Widget.inherit({
 
     _renderTemplates: function(templatesInfo) {
         const that = this;
-        each(templatesInfo, function(_, info) {
+
+        let itemsCountWithLabelTemplate = 0;
```
Comment (ksercs) [inline js/ui/form/ui.form.layout_manager.js]:
```suggestion
        let itemsWithLabelTemplateCount = 0;
```

File: js/ui/form/ui.form.layout_manager.js [hunk 5]
```diff
@@ -560,6 +570,15 @@ const LayoutManager = Widget.inherit({
         }
 
         const name = item.dataField || item.name;
+        const formOrLayoutManager = this._getFormOrThis();
+
+        const onLabelTemplateRendered = () => {
+            this._incTemplateRenderedCallCount();
+
+            if(formOrLayoutManager.option('templatesRenderAsynchronously') && this._shouldAlignLabelsOnTemplateRendered(itemsCountWithLabelTemplate)) {
```
Comment (ksercs) [inline js/ui/form/ui.form.layout_manager.js]: maybe it's metter to move `formOrLayoutManager.option('templatesRenderAsynchronously')` to `_shouldAlignLabelsOnTemplateRendered` to make this code clearer?

File: js/ui/form/ui.form.layout_manager.js [hunk 6]
```diff
@@ -598,6 +618,14 @@ const LayoutManager = Widget.inherit({
         });
     },
 
+    _incTemplateRenderedCallCount() {
+        this._labelTemplateRenderedCall = (this._labelTemplateRenderedCall ?? 0) + 1;
```
Comment (ksercs) [inline js/ui/form/ui.form.layout_manager.js]:
```suggestion
        this._labelTemplateRenderedCallCount = (this._labelTemplateRenderedCall ?? 0) + 1;
```

File: js/ui/form/ui.form.layout_manager.js (hunk 5 above)
Comment (EugeniyKiyashko) [reply js/ui/form/ui.form.layout_manager.js]: LGTM

**Review by ksercs — APPROVED** [review]

**Pull request merged**